gcc/rust/lang.opt

```diff
@@ -38,6 +38,18 @@ Wall
 Rust
 ; Documented in c.opt
 
+Wunused-variable
+Rust Var(warn_unused_variable) Init(1) Warning
+; documented in common.opt
+
+Wunused-const-variable
+Rust Warning Alias(Wunused-const-variable=, 2, 0)
+Warn when a const variable is unused.
+
+Wunused-const-variable=
+Rust Joined RejectNegative UInteger Var(warn_unused_const_variable) Init(1) Warning LangEnabledBy(Rust,Wunused-variable, 1, 0) IntegerRange(0, 2)
+Warn when a const variable is unused.
```

Comment:
I'm back to working on resolving <https://gcc.gnu.org/PR105914> "gccrs setting warn_unused_variable breaks thousands of non-Rust tests".  I do understand what the problem is (details to be shared later), and how to resolve it.  And, I do have a set of patches for GCC options machinery, to be upstreamed separately, to make it less likely that the same issue appears again.

I do understand that you'd like `-Wunused-variable`, `-Wunused-const-variable=1` enabled by default.  (As implemented here, via `Init(1)`.)  Now I need to know: (1) should that be enforced unconditionally (that is, no way to disable those diagnostics), or (2) should it be possible to specify `-Wno-unused-variable` etc. to disable the diagnostics?  (Natural for GCC would be the latter, (2).)